tsdb/engine/tsm1/array_cursor.gen.go

```diff
@@ -190,6 +190,8 @@ func (c *floatArrayDescendingCursor) reset(seek, end int64, cacheValues Values,
 		})
 		if c.cache.pos == len(c.cache.values) {
 			c.cache.pos--
+		} else if c.cache.values[c.cache.pos].UnixNano() == seek {
+			c.cache.pos--
```

Comment:
Another area of code I'm unfamiliar with. What exactly is the purpose of the array cursor type? And why is this `else if` now necessary?

Reply:
`last()` pushdowns use descending array cursors. If points fall on the boundary of the passed in range, we want to keep the points on the left hand boundary and exclude the points on the right hand boundary. This line concerns the right hand boundary. `seek` represents the `_stop` time. So if we see a point with timestamp equal to `_stop` we want to exclude it.